nonce/nonce.go

```diff
@@ -117,11 +138,25 @@ func (ns *NonceService) encrypt(counter int64) (string, error) {
 	ct := ns.gcm.Seal(nil, nonce, pt, nil)
 	copy(ret, nonce[4:])
 	copy(ret[8:], ct)
-	return base64.RawURLEncoding.EncodeToString(ret), nil
+
+	return ns.prefix + base64.RawURLEncoding.EncodeToString(ret), nil
 }
 
 func (ns *NonceService) decrypt(nonce string) (int64, error) {
-	decoded, err := base64.RawURLEncoding.DecodeString(nonce)
+	var decoded []byte
+	if ns.prefix != "" {
+		var prefix string
+		var err error
+		prefix, nonce, err = splitNonce(nonce)
```

Comment:
I'd prefer not to rely on overwriting the input parameter `nonce` with the body after splitting, since that makes it a little more complicated to reason about what value this variable has at what point. Can you choose a new name, like `body`?

Reply:
I'm not sure what you mean here, we need to overwrite nonce so that when we decode after this block we don't attempt to decode the prefix as well, or do you want to overwrite in a separate step with a different variable receiving the actual nonce part here?

Reply:
I mean, assign the nonce body to `body`, then decode the body. You'll need to declare `body` outside the `if` scope, naturally.